Adapter.C

```diff
@@ -923,6 +924,69 @@ void preciceAdapter::Adapter::setupCheckpointing()
+    toRemove.clear();                                                                                                 \
+    /* Iterate through fields in OpenFOAM registry */                                                                 \
+    for (const word& obj : mesh_.sortedNames<GeomField>())                                                            \
+    {                                                                                                                 \
+        regFields.push_back(obj);                                                                                     \
+    }                                                                                                                 \
+    /* Iterate through checkpointed fields */                                                                         \
+    for (uint i = 0; i < GeomFieldCopies_.size(); i++)                                                                \
+    {                                                                                                                 \
+        found = false;                                                                                                \
+        obj = GeomFieldCopies_.at(i)->name();                                                                         \
+        for (uint j = 0; j < regFields.size(); j++)                                                                   \
+        {                                                                                                             \
+            if (obj == regFields.at(j))                                                                               \
+            {                                                                                                         \
+                found = true;                                                                                         \
+                break;                                                                                                \
+            }                                                                                                         \
+        }                                                                                                             \
+        if (!found)                                                                                                   \
+        {                                                                                                             \
+            toRemove.push_back(obj);                                                                                  \
+        }                                                                                                             \
+    }                                                                                                                 \
```

Comment:
```suggestion
    /* Iterate through checkpointed fields */                                                                         
  for (const auto& geomFieldCopy : GeomFieldCopies_) {                                  
     auto obj = geomFieldCopy->name();
     if (std::find(regFields.begin(), regFields.end(), obj) == regFields.end()) {
         toRemove.push_back(obj);
     }
 }                                                                                              
```
Maybe an STL version of the code would be more readable?

Reply:
I agree this is more readable and I even had this approach first, but then I would have to include the 'algorithms' header. Is that alright? It's not used elsewhere.

Reply:
Including the algorithms header is fine, and we should be using the STL wherever it makes sense. Here, it indeed does, as there was already some uneasiness on the index-based loops.

Generally, the fact that there is not so much STL in the code is not by design. Feel free to suggest better implementations wherever you see the potential. The only thing we need to be aware is keeping the C++ standard compatible with the preCICE and OpenFOAM requirements. C++14 should be safe and this is defined by the WMake rules in the respective OpenFOAM installation (`/usr/lib/openfoam/openfoam2406/wmake/rules/General/Gcc/c++`). They [announced](https://www.openfoam.com/news/main-news/openfoam-v2312) moving to C++17 at some point in 2024, could be in the upcoming v2412.

Reply:
Thanks for the reply! I will then quickly refactor and test the code.